docs/modules/ROOT/pages/references/style-guide.adoc

```diff
@@ -0,0 +1,273 @@
+= Style Guide
+
+The Jsonnet related parts of this guide build on top of https://github.com/databricks/jsonnet-style-guide/blob/30d4d54444c1001ba9561afd7c8de405630a2ca7/README.md[Jsonnet Guide] by https://databricks.com/[Databricks].
+
+http://creativecommons.org/licenses/by-nc-sa/4.0/[image:https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png[Creative Commons License]] +
+This work is licensed under a http://creativecommons.org/licenses/by-nc-sa/4.0/[Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License].
+
+== General Style
+
+* Encode text files with UTF-8.
+* Use Unix line endings (lf).
+* Always end a file with a single empty line.
+  This doesn't apply to empty files (for example `.gitkeep`).
+* Use two spaces for indentation except for `Makefile` where a single tab is to be used.
+* Trim whitespaces from the end of a line.
+
+[NOTE]
+====
+Repositories of Project Syn provide an https://editorconfig.org/[EditorConfig].
+When using a https://editorconfig.org/#download[supported editor], space types and number of spaces will automatically be done correctly.
+====
+
+== Asciidoc Style
+
+* Don't split senctences across lines but keep them on one single line.
+* Start a new line for each and every sentence.
+* Follow the https://docs.microsoft.com/en-us/style-guide/welcome/[Microsoft Writing Style Guide].
+
+[NOTE]
+====
+The Microsoft Writing Style is being enforced with https://github.com/errata-ai/vale[Vale].
+The implementation used can be found at https://github.com/errata-ai/Microsoft[errata-ai/Microsoft].
+====
+
+== Jsonnet Style
+
+== Syntactic Style
+
+=== Auto Formatting
+
+Use `jsonnet fmt` to format files. This will fix basic style errors.
+
+=== Variable Declaration
+
+* Variables should be named in `camelCase` style, and should have self-evident names.
++
+[source,jsonnet]
+----
+local serverPort = 1000;
+local clientPort = 2000;
+----
+
+* Prefer `local` to `::` syntax for private/local variables.
+  Unlike `::`, variables defined with `local` can't be overridden by children, nor accessed by other files.
++
+[source,jsonnet]
+----
+{
+  // CORRECT
+  local myVariable = 3,
```

Comment:
I'm not sure what the point of having a local inside an object is, contrary to having a private field this doesn't give any extra functionality over a local outside the object, or just directly using the value of the local.

I would do this as

```
local myVariable = 3;
{
  result: myVariable + 1,
}
```

Reply:
Agreed, functionally those are the same. That is except for the different scope of the variable definition. That is probably something we could discuss for the best practices guide.

Reply:
True enough, I haven't really found a need for having complex scoping in a single file so far.